Handle repo redirect

JuanitoFatas opened the pull request:
When repo changed to a new location, the response contains a "Moved Permanently" message
and the new repo location url, we follow the url and get the (file) contents to find changelog-like file.

winston commented on lib/whatsnew/remote_files.rb:
```diff
@@ -44,19 +44,41 @@ def client
+        resources = client.contents(repo)
+
+        if resources.respond_to?(:message) && resources.message == "Moved Permanently" &&
+           new_location = resources.url
```
Indentation?

JuanitoFatas replied:
It is part of the coditional.

winston commented on lib/whatsnew/remote_files.rb:
```diff
@@ -44,19 +44,41 @@ def client
+      end
+
+      def http_get(uri)
+        Dish(JSON.parse(uri.read))
```
Would prefer not to pull in another dependency just for `resources.name`.

Also, the extraction of `normalize_uri` doesn't yield more readability than just doing 
`JSON.parse(URI(uri).read)`

JuanitoFatas replied:
> Also, the extraction of normalize_uri doesn't yield more readability than just doing 
JSON.parse(URI(uri).read)

Well, I was thinking if I want to do more normalizations, and if that is going to happen, there will be the need for putting in a normalize_uri method, but for now I only want to keep it simple.

winston reviewed (changes requested):
The specs are failing?

JuanitoFatas committed:
Handle repo redirects, search in new repo

If repo moved to a new location, we search the new repo’s contents for
changelog.

JuanitoFatas committed:
Use Net::HTTP.get instead

JuanitoFatas committed:
Add a changelog
